backend-api/src/functions/tests/pact/asyncCredential/asyncCredentialStateConfiguration.ts

```diff
@@ -0,0 +1,78 @@
+import { Dependencies } from "../../../asyncCredential/handlerDependencies";
+import { registeredLogs } from "../../../asyncCredential/registeredLogs";
+import { MockEventWriterSuccess } from "../../../services/events/tests/mocks";
+import { Logger } from "../../../services/logging/logger";
+import { MockLoggingAdapter } from "../../../services/logging/tests/mockLogger";
+import {
+  MockClientRegistryServiceGetPartialClientSuccessResultIPV,
+  MockSessionServiceCreateSessionSuccessResult,
+  MockTokenServiceInvalidSignatureIPV,
+  MockTokenServiceSuccessIPV,
+} from "../../../testUtils/asyncCredentialMocks";
+
+const env = {
+  SIGNING_KEY_ID: "mockKid",
+  ISSUER: "mockIssuer",
+  SESSION_TABLE_NAME: "mockTableName",
+  SESSION_TABLE_SUBJECT_IDENTIFIER_INDEX_NAME: "mockIndexName",
+  SESSION_TTL_IN_MILLISECONDS: "12345",
+  SQS_QUEUE: "mockSqsQueue",
+  CLIENT_REGISTRY_PARAMETER_NAME: "mockParmaterName",
+};
+
+export class AsyncCredentialStateConfiguration {
```

Comment:
I think we could really simplify this -> one idea could be:

```
getDependenciesbyScenario(scenario: ScenarioType): IDependenices => {

switch(scenario) { 
   case 'INVALID_TOKEN': { 
      return {
tokenService: () => new MockInvalidTokenService
      break; 
   } 
   default: { 
      return {
tokenService: () => new MockTokenService()
      }
  }
}
   


```

Reply:
ignoring formatting/syntax - but hopefully you get the gist

Reply:
I've updated the function on this class to set the dependencies by scenario. Happy to discuss further if needed

Reply:
Update: Sonar doesn't like switch statements with fewer than 3 cases so I've changed the functions for to use if statements. Maybe something to review when more cases are added.